### PR TITLE
feat(formatter): add formatter

### LINKED ISSUE #143: support document formatting request
See https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_formatting

AC: As a user I can format my flux script using the language server.

### PR DESCRIPTION
**Pull request by kelwang**
Closes #143

Add the formatter

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [x] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: tests/handlers.rs [hunk 1]
```diff
@@ -471,14 +471,14 @@ speculate! {
 
                 labels.sort_by(|a, b| a.to_lowercase().cmp(&b.to_lowercase()));
 
-                assert_eq!(labels, vec!["csv", "file"], "returns correct items");
+                assert_eq!(labels, vec!["csv", "file", "url"], "returns correct items");
```
Comment (kelwang) [inline tests/handlers.rs:474]: this is due to flux version update

File: src/format/mod.rs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                        line_number - context_queue.len() as u32,
+                        line_number_orig - context_queue.len() as u32,
+                    );
+                }
+
+                while let Some(line) = context_queue.pop_front() {
+                    mismatch
+                        .lines
+                        .push(DiffLine::Context(line.to_owned()));
+                }
```
Comment (jsteenb2) [inline src/format/mod.rs:121]: curiousity: would it be possible to create a closure around this code to take the `str` arg and do the stuff here?

File: src/handlers/document_formatting.rs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                Response::new(prequest.base_request.id, Some(msg));
+
+            let json = match response.to_json() {
+                Ok(s) => Ok(s),
+                Err(_) => Err(String::from(
+                    "Failed to serialize formatting response",
+                )),
+            };
+
+            return Ok(Some(json?));
```
Comment (jsteenb2) [inline src/handlers/document_formatting.rs:40]: curiosity: what is happening here if json returns an error? will be some OK that wraps an error, or will it return nothing? 

trying to grok the `json?` piece

Comment (kelwang) [reply src/handlers/document_formatting.rs:40]: i copied from our existing code. but it's very unlikely something bad can happen here

File: src/format/mod.rs (hunk 2 above)
Comment (kelwang) [reply src/format/mod.rs:121]: let me try to make it a little simpler

**Pull request closed without merging**